solutions/silver/usaco-1159.mdx

```diff
@@ -13,6 +13,104 @@ author: Nathan Gong, Jesse Choe
 
 <LanguageSection>
 
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+using ll = long long;
+using vi = vector<int>;
+
+void solve() {
+	ll n, m;
+	cin >> n >> m;
+
+	// Adjacency list
+	vector<vector<ll>> adj(n);
+	for (int i = 0; i < m; ++i) {
+		ll u, v;
+		cin >> u >> v;
+		--u;
+		--v;  // Convert to 0-based indexing
+		adj[u].push_back(v);
+		adj[v].push_back(u);
+	}
+
+	// Connected components parent and count
+	vector<ll> vis(n, -1);
```

Comment:
(optional)
naming this array `color` or `id` might be more appropriate if you're labeling connected components